contracts/multiswap-contracts/FiberRouter.sol

```diff
@@ -619,444 +514,144 @@ function _swapOnSameNetwork(
-     * @param oneInchData The data containing information for the 1inch swap
-     * @return returnAmount The amount of tokens received after the swap and transaction execution
+     * @param toToken The address of the output token from the swap
+     * @param amountIn The exact amount of input tokens to be swapped
+     * @param minAmountOut The minimum amount of output tokens expected after the swap
+     * @param router The router address
+     * @param data The calldata for the swap
      */
-    function handleSwap(
-        address payable to,
+    function _swapAndCheckSlippage(
+        address recipient,
         address fromToken,
+        address toToken,
         uint256 amountIn,
-        uint256 amountOut,
-        bytes memory oneInchData
-    ) internal returns (uint256 returnAmount) {
-        // Decoding oneInchData to get the required parameters
-        (
-            address executor,
-            SwapDescription memory desc,
-            bytes memory permit,
-            bytes memory swapData
-        ) = abi.decode(
-            oneInchData,
-            (address, SwapDescription, bytes, bytes)
-        );
-        // Manually create a new SwapDescription for IOneInchSwap
-        IOneInchSwap.SwapDescription memory oneInchDesc = IOneInchSwap
-            .SwapDescription({
-                srcToken: IERC20(desc.srcToken),
-                dstToken: IERC20(desc.dstToken),
-                srcReceiver: desc.srcReceiver,
-                dstReceiver: desc.dstReceiver,
-                amount: desc.amount,
-                minReturnAmount: desc.minReturnAmount,
-                flags: desc.flags
-            });
-
-        // Accessing fields of the desc instance of SwapDescription struct
-        require(
-            to == desc.dstReceiver,
-            "FR: recipient address bad oneInch Data"
-        );
-        require(amountIn == desc.amount, "FR: inputAmount bad oneInch Data");
-        require(
-            amountOut == desc.minReturnAmount,
-            "FR: outAmount bad oneInch Data"
-        );
-        require(fromToken == desc.srcToken, "FR: srcToken bad oneInch Data");
-
-        // Additional safety check
-        require(oneInchData.length >= 4, "Data too short for valid call");
-
-        // Performing the swap
-        ( returnAmount,) = IOneInchSwap(oneInchAggregatorRouter).swap(
-            executor,
-            oneInchDesc,
-            permit,
-            swapData
-        );
-        emit SwapHandled(
-            oneInchAggregatorRouter,
-            to,
-            fromToken,
-            amountIn,
-            returnAmount // should be returned 
-        );
+        uint256 minAmountOut,
+        address router,
+        bytes memory data
+    ) internal returns (uint256) {
+        require(routerAllowList[_getKey(router, data)], "FR: Router and selector not whitelisted");
+        _approveRouter(fromToken, router, amountIn);
```

Comment:
If we can guarantee that there are no funds in the router outside the context of a swap, we can potentially do a blanket approval and save some gas. The gas saving will be small though, so I am not sure if it is worth it